src/enigma2/data/ChannelGroup.h

```diff
@@ -33,9 +35,20 @@ namespace enigma2
 {
   namespace data
   {
+    class ChannelGroup;
+    typedef std::shared_ptr<enigma2::data::ChannelGroup> ChannelGroupPtr;
```

Comment:
These kind of typedefs always have the problem that you can't forward declare the resulting type and need either to include the corresponding header or to copy the typedef (as you have already done in Channel.h). I would avoid that and always use the full type.

Reply:
Done